### PR TITLE
Fix anti join w/ filter that is true or failing for non-matching rows

### PR DESCRIPTION
**Pull request by mbasmanova**
Anti join returned incorrect result or failed when extra filter passed or failed on rows without a match.

The fix is to not evaluate the filter on rows without a match and always include these in the join results.

A similar change needs to be applies in a follow-up PR to the left join. The
filter should not be evaluated on rows without a match to avoid failing the
query unnecessarily.

See #3555 for addition context.

### CODE REVIEW

File: velox/exec/HashProbe.cpp [hunk 1]
```diff
@@ -1145,13 +1145,32 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
...
   auto* rawOutputProbeRowMapping =
       outputRowMapping_->asMutable<vector_size_t>();
 
-  fillFilterInput(numRows);
   filterInputRows_.resizeFill(numRows);
 
+  // For anti join, do not evaluate filter on rows without a match. Evaluating
+  // filter on rows without a match may trigger errors in filter evaluation and
+  // fail the query unnecessarily.
+  // TODO Apply to the same to left joins.
```
Comment (xiaoxmeng) [inline velox/exec/HashProbe.cpp:1158]: nit:
```
// ... comment ...
//
// TODO Apply to the same to left joins.
```

File: velox/exec/HashProbe.cpp [hunk 2]
```diff
@@ -1145,13 +1145,32 @@ int32_t HashProbe::evalFilter(int32_t numRows) {
...
+  if (isAntiJoin(joinType_)) {
```
Comment (Yuhta) [inline velox/exec/HashProbe.cpp:1159]: Should we do the same to null-aware anti join?

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova nice change and analysis!

File: velox/exec/HashProbe.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/exec/HashProbe.cpp:1159]: That's a good question. I have an AI to look into that.

**Pull request closed without merging**